drivers/timer/cortex_m_systick.c

```diff
@@ -21,9 +18,25 @@ void z_ExcExit(void);
 #define MAX_TICKS ((COUNTER_MAX / CYC_PER_TICK) - 1)
 #define MAX_CYCLES (MAX_TICKS * CYC_PER_TICK)
 
+/* Minimum cycles in the future to try to program.  Note that this is
+ * NOT simply "enough cycles to get the counter read and reprogrammed
+ * reliably" -- it becomes the minimum value of the LOAD register, and
+ * thus reflects how much time we can reliably see expire between
+ * calls to elapsed() to read the overflow bit.  So it needs to be set
+ * to be larger than the maximum time the interrupt might be masked.
+ * Choosing a fraction of a tick is probably a good enough default,
+ * with an absolute minimum of 1k cyc.
+ */
+#define MIN_DELAY MAX(1024, (CYC_PER_TICK/16))
```

Comment:
I think this is a good approach. It won't suffer from a high tick resolution due to the minimum 1024 cycles.

Reply:
On the other hand, it seems to me we lose accuracy for really fast platforms (e.g. SAM E70, 300MHz) - we are truncated to 1/16th of a tick that is usually not below 1msec...

Reply:
There's no change in timer accuracy, as ticks are always aligned.  This changes the threshold in the future at which a timer interrupt can be scheduled (if one isn't already scheduled).  There has to be some non-zero threshold there, though I won't try to argue that this is the perfect default.